packages/flutter_tools/lib/executable.dart

```diff
@@ -159,3 +171,17 @@ String _doctorText() {
     return 'encountered exception: $error\n\n${trace.toString().trim()}\n';
   }
 }
+
+Future<Null> _exit(int code) async {
+  if (flutterUsage.enabled) {
+    Stopwatch stopwatch = new Stopwatch()..start();
+    await flutterUsage.ensureAnalyticsSent();
+    printTrace('ensureAnalyticsSent: ${stopwatch.elapsedMilliseconds}ms');
+  }
+
+  logger.flush();
+
+  await Timer.run(() {
```

Comment:
add a comment here explaining the hoops we're jumping through, in case someone looks at this code and cargo-cults it...